### ISSUE TITLE
Solar panels not working after installation

### ISSUE THREAD

**Issue by Bulletfang** [CleverRaven/Cataclysm-DDA#28514]
**Describe the bug**  
After installing standard Solar Panels on an electric vehicle, batteries are not charged up anymore, even if the vehicle spawned with Solar Panels that previously worked.

**To Reproduce**  
1. Find electric vehicle (Electric Sports Car and normal Electric Car in my specific case).
2. Install Solar Panels (whether there are properly working Solar Panels already installed or not does not influence this bug).
3. Batteries do not receive power, regardless of weather conditions.

**Expected behavior**  
Batteries should receive power in daylight.

**Versions and configuration(please complete the following information):**
 - OS: [Windows 10]
 - Game Version: [0.C-37588-gc5f2b3b]
 - Graphics version: [Tiles]
 - Mods loaded: [Dark Days Ahead, Disable NPC Needs, Filthy Clothing, Simplified Nutrition, Just Craftable Gunpowder, Makeshift Items Mod, PKs Rebalancing, Alternative Map Key, Mutant NPCs, More Locations, Fuji's More Buildings, Urban Development, PK - No Doom Monsters, Nonperishable Overhaul, StatsThroughSkills, DinoMod]

**Additional context**  
Savegame file for reproduction (I manually renamed the file ending from .rar to .zip, in case there are problems with the file). 
[save.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/2936521/save.zip)

edit: Installing the same Solar Panels on a gasoline driven vehicle will charge the batteries as one would expect.

**Comment by davidpwbrown** [comment]
Enable debug mode ( ~ ) in-game.
Wait some amount of time, look at the message log to see if it says "vehicle got x amount of power from solar panels"
with debug mode on , the vehicle interaction UI should show how long to vehicle is charged.
If weather is not sunny, then charging will be very slow.
I don't use all those mods, so that savegame would take too long for me to test, but on my build of latest master, solar panels are working fine.

**Comment by mlangsdorf** [comment]
There's a unit test that runs with every build that tests whether solar panels work.  It hasn't failed recently.

I just loaded a game with my standard test character.  I debug spawned a light frame, a solar panel, and a car battery.  I started vehicle construction, added the battery and solar panel, and then waited for 30 minutes on a cloudy day on the first day of summer.  As expected, the battery had recharged 4%.

Can't confirm, closing.
